### PR TITLE
FOGL-9028 Add automated ingest rate monitoring to the south service

### PR DESCRIPTION
**Pull request by MarkRiddoch**
@Mohit04tomar If we ran the current system tests against this branch would we see any alerts raised or do we need to modify the test to capture them?

### CODE REVIEW

**Comment by Mohit04tomar** [comment]
@MarkRiddoch 

> If we ran the current system tests against this branch would we see any alerts raised

No, we will not receive any alerts because when the alert feature was introduced via FOGL-8439, no JIRA ticket was filed to create a system test.

> do we need to modify the test to capture them?


We need a JIRA with all the required information to create a system test for end-to-end (E2E) testing of alerts. 
Currently, we only have API integration tests for alerts but no E2E tests.

**Review by pintomax — APPROVED** [review]

File: C/services/south/ingestRate.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+/**
+ * Destructor for the ingest rate class
+ */
+IngestRate::~IngestRate()
+{
+}
+
+/**
+ * Update the configuration of the ingets rate mechanism
```
Comment (cyberwalk3r) [inline C/services/south/ingestRate.cpp:35]: *ingest

File: C/services/south/ingest.cpp [hunk 2]
```diff
@@ -1153,7 +1181,7 @@ void Ingest::configChange(const string& category, const string& newConfig)
 }
 
 /**
- * Return the numebr fo queued readings in the south service
+ * Return the number fo queued readings in the south service
```
Comment (cyberwalk3r) [inline C/services/south/ingest.cpp:1184]: *of

File: docs/tuning_fledge.rst [hunk 3]
```diff
@@ -182,6 +187,32 @@ When collection is enabled the following counters will be collected for the sout
...
+
++---------------+
+| |south_alert| |
++---------------+
+
+The algorithm requires two consecutive out of range ingest rates to prevent the alert trigger for an isolated peak or trough in data collection caused by a one off action occurring on the host platform, or within Fledge. If in a subsequent monitoring period the flow rate returns to acceptable limits, the alert in the status bar will be cleared.
+
+.. note::
+
+   This ingest rate monitoring is designed to be applicable is as many situations as possible. There are however some cases in which this monitoring will create false reports of issues. This may be able to be reduced or eliminated by using the tuning options, but this may not be true in all cases. In particular an asynchronous south plugin that reports data at unpredictable time intervals will most likely not be suitable for this type of monitoring and the monitoring should be disabled by setting a 0 monitoring interval.
```
Comment (cyberwalk3r) [inline docs/tuning_fledge.rst:215]: applicable *in as many situations as possible.

**Review by cyberwalk3r — APPROVED** [review]

**Review by pintomax — APPROVED** [review]

**Pull request merged**